### PR TITLE
fix: Typos in SQSScheduledExecutorService delay math

### PR DESCRIPTION
**Pull request by robin-aws**
*Issue #, if available:*

- #28

*Description of changes:*

Also improved test coverage for this area via Awaitility, specifically setting a minimum wait time for tasks to execute, since the effect of this bug was to trigger recurring tasks too early.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/sqs/SQSScheduledExecutorService.java [hunk 1]
```diff
@@ -122,8 +122,8 @@ public int compareTo(Delayed other) {
         public SendMessageRequest toSendMessageRequest() {
             SendMessageRequest request = super.toSendMessageRequest();
 
-            int sqsDelaySeconds = (int)Math.min(TimeUnit.NANOSECONDS.toSeconds(delay), MAX_SQS_DELAY_SECONDS);
-            if (sqsDelaySeconds < 0) {
+            int sqsDelaySeconds = (int)Math.min(TimeUnit.NANOSECONDS.toSeconds(Math.abs(delay)), MAX_SQS_DELAY_SECONDS);
```
Comment (haocheng-aws) [inline src/main/java/com/amazonaws/services/sqs/SQSScheduledExecutorService.java]: nit: How about use `Math.max(1, sqsDelaySeconds)`?

Comment (robin-aws) [reply src/main/java/com/amazonaws/services/sqs/SQSScheduledExecutorService.java]: I like that, thanks!
